File: python/Vehicle.py
# Vehicle : plat nomor, merk, tahun produksi, warna
class Vehicle:
    __plat_nomor = ""
    __merk = ""
    __tahun_produksi = ""
    __warna = ""

    # contructor
    def __init__(self, plat_nomor, merk, tahun_produksi, warna):
        self.__plat_nomor = plat_nomor
        self.__merk = merk
        self.__tahun_produksi = tahun_produksi
        self.__warna = warna

    # getter setter
    def getPlat(self):
        return self.__plat_nomor

    def setPlat(self, plat):
        self.__plat_nomor = plat

    def getMerk(self):
        return self.__merk

    def setMerk(self, merk):
        self.__merk = merk

    def getTahunProduksi(self):
        return self.__tahun_produksi

    def setTahunProduksi(self, tahun_produksi):
        self.__tahun_produksi = tahun_produksi

    def getWarna(self):
        return self.__warna

    def setWarna(self, warna):
        self.__warna = warna

    # methods of all vehicle

    def start():
        print("Broom Broom")

    def stop():
        print("Stopping")

File: cpp/Vehicle.cpp
#pragma once
#include <iostream>
#include <string>

using namespace std;

class Vehicle {
 private:
  string plat_nomor;
  string merk;
  string tahun_produksi;
  string warna;

 public:
  // construktor kosong
  Vehicle() {}

  Vehicle(string plat, string merk, string tahun, string warna) {
    this->plat_nomor = plat;
    this->merk = merk;
    this->tahun_produksi = tahun;
    this->warna = warna;
  }

  // setter getter
  string getPlat() { return this->plat_nomor; }
  void setPlat(string plat) { this->plat_nomor = plat; }

  string getMerk() { return this->merk; }
  void setMerk(string merk) { this->merk = merk; }

  string getTahunProduksi() { return this->tahun_produksi; }
  void setTahunProduksi(string tahun) { this->tahun_produksi = tahun; }

  string getWarna() { return this->warna; }
  void setWarna(string warna) { this->warna = warna; }
  virtual void printDetails() {
    cout << "Plat Nomor: " << getPlat() << endl;
    cout << "Merk: " << getMerk() << endl;
    cout << "Tahun_produksi: " << getTahunProduksi() << endl;
    cout << "Warna: " << getWarna() << endl;
  }
  ~Vehicle() {}
};
